api/src/main/java/org/hawkular/apm/api/model/trace/Node.java

```diff
@@ -231,6 +250,49 @@ public Node setDetails(Map<String, String> details) {
     }
 
     /**
+     * This method determines whether there is atleast one
+     * property with the supplied name.
+     *
+     * @param name The property name
+     * @return Whether a property of the supplied name is defined
+     */
+    public boolean hasProperty(String name) {
+        for (Property property : this.properties) {
+            if (property.getName().equals(name)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * This method returns the set of properties having the
+     * supplied property name.
+     *
+     * @param name The property name
+     * @return The set of properties with the supplied name
+     */
+    public Set<Property> getProperties(String name) {
+        Set<Property> ret = new HashSet<Property>();
```

Comment:
java 8: return `this.properties.stream().filter(property -> property.getName().equals(name)).collect(Collectors.toSet());`